src/frontend/api.ts

```diff
@@ -143,9 +152,113 @@ export const SERVER_START_TIMEOUT = 30000;
 
 const pixelRatio = PixelRatio.get();
 
-interface ApiParam {
-  baseUrl: string;
-  timeout?: number;
+function createRequestClient({
+  baseUrl,
+  onReady,
+}: {
+  baseUrl?: string;
+  onReady?: () => Promise<void>;
+} = {}) {
+  const req = ky.extend({
+    prefixUrl: baseUrl,
+    timeout: false,
```

Comment:
there's no manual timeout code for the map server like there is for the app server at the moment, meaning that there's nothing we do to handle the case where the map server hangs. solution is to either:

- specify a timeout (i.e. let that be an option for the function)
- add proper timeout handling code specific to map server

Reply:
might be a candidate to follow-up on after fixing up the server start/stop code, since it'll probably be related to that